matter_server/server/device_controller.py

```diff
@@ -417,47 +443,67 @@ async def _call_sdk(self, func: Callable[..., _T], *args: Any, **kwargs: Any) ->
             ),
         )
 
-    async def _check_interviews(self) -> None:
-        """Check if any nodes need to be (re)interviewed."""
-        assert self.server.loop is not None
-        # Reinterview all nodes that have an outdated schema
-        # and/or have been interviewed more than 30 days ago.
-        for node in self._nodes.values():
+    async def _check_subscriptions_and_interviews(self) -> None:
+        """Setup subscriptions (and interviews) for known nodes."""
+        for node_id, node in self._nodes.items():
+            # (re)interview node (only) if needed
             if (
-                SCHEMA_VERSION > node.interview_version
+                node is None
+                or node.interview_version < SCHEMA_VERSION
                 or (datetime.utcnow() - node.last_interview).days > 30
             ):
-                await self.interview_node(node.node_id)
+                try:
+                    await self.interview_node(node_id)
+                except NodeInterviewFailed as err:
+                    LOGGER.warning(
+                        "Unable to interview Node %s, we will retry later in the background.",
+                        node_id,
+                        exc_info=err,
+                    )
+                    continue
+
+            # setup subscriptions for the node
+            if node_id not in self._subscriptions:
+                # If the node is unreachable on the network now,
+                # it will throw a NodeNotResolving exception, catch this,
+                # log this and just try to resolve this node in the next run.
+                try:
+                    await self.subscribe_node(node_id)
+                except NodeNotResolving as err:
+                    LOGGER.warning(
+                        "Unable to contact Node %s,"
+                        " we will retry later in the background.",
+                        node_id,
+                        exc_info=err,
+                    )
+                    continue
+
         # reschedule self to run every hour
-        self.server.loop.call_later(3600, self._schedule_interviews)
+        def _schedule():
+            asyncio.create_task(self._check_subscriptions_and_interviews())
 
-    def _schedule_interviews(self) -> None:
-        """Schedule interviews."""
-        asyncio.create_task(self._check_interviews())
+        # NOTE: do not do this in one call otherwise asyncio will start
```

Comment:
Is this note really necessary? That's how asyncio works 😄

Reply:
hahaha, well, I've seen the mistake happen a lot.